modules/reitit-core/src/reitit/core.cljc

```diff
@@ -39,14 +39,14 @@
      (walk-one [pacc macc routes]
        (if (vector? (first routes))
          (walk-many pacc macc routes)
-         (let [[path & [maybe-arg :as args]] routes
-               [data childs] (if (vector? maybe-arg)
-                               [{} args]
-                               [maybe-arg (rest args)])
-               macc (into macc (expand data opts))]
-           (if (seq childs)
-             (walk-many (str pacc path) macc childs)
-             [[(str pacc path) macc]]))))]
+         (if (string? (first routes))
```

Comment:
Maybe when instead? I at least instinctively look for the third argument in ifs.

Reply:
There is no consensus on this: `when` has implicit `do` for side-effects, so one needs to look for the last statement in the block to see what it returns and see if it side-effects. `if` always just returns.

But, changed this to `when`.

related: https://github.com/jonase/kibit/issues/2